internal/oci/container.go

```diff
@@ -267,14 +277,19 @@ func (c *Container) Metadata() *pb.ContainerMetadata {
 
 // State returns the state of the running container
 func (c *Container) State() *ContainerState {
-	c.opLock.RLock()
-	defer c.opLock.RUnlock()
-	return c.state
+	c.stateLock.RLock()
+	defer c.stateLock.RUnlock()
+	if c.state != nil {
+		cs := *c.state
+		return &cs
```

Comment:
I think we just return ContainerState instead of a pointer from this function.

Reply:
There are 36 references to Container#State in the code base.
Do you mind if we leave this refactoring to the next PR ?

Reply:
Sure!